onchain/aiken/validators/fraud-proof/min-fee/step.ak

```diff
@@ -0,0 +1,111 @@
+  get_singleton_ct_token, get_tx_root_validate_block_hash, handle_cancel,
+  validate_final_step_output, validate_membership_proof,
+} as fraud_proof_utils
+use midgard/fraud_proof/min_fee/step.{
+  Cancel, Continue, StepRedeemer as Step1Redeemer,
+}
+use midgard/ledger_state.{MidgardTxCompact}
+
+validator main(
+  // Fraud proof token policy
+  fraud_proof_token_policy: PolicyId,
+  // Script hash of stake validator (forwarded merkle proof logic)
+  plutarch_midgard_merkle_validator_inclusion: ScriptHash,
+  // Computation thread token policy
+  ct_token_policy_id: PolicyId,
+) {
+  spend(
+    datum: Option<StepDatum>,
+    redeemer: Step1Redeemer,
+    input: OutputReference,
+    tx: Transaction,
+  ) {
+    expect Some(step_datum) = datum
+    expect Some(own_input) =
+      list.find(tx.inputs, fn(a_input) { a_input.output_reference == input })
+
+    when redeemer is {
+      Cancel { index_output } -> {
+        expect
+          handle_cancel(
+            own_input,
+            index_output,
+            step_datum,
+            ct_token_policy_id,
+            tx,
+          )
+        True
+      }
+      // Step 1: Membership proof for tx1
+      Continue {
+        bad_state_queue_node_ref_input_index,
+        bad_tx_hash,
+        bad_tx_info_hash,
+        bad_tx_to_root_proof_path,
+        bad_tx,
+        bad_tx_body,
+        index_output,
+        computation_thread_redeemer_index,
+      } -> {
+        let ct_token_asset_name =
+          get_singleton_ct_token(own_input, ct_token_policy_id)
+        let extracted_tx_root =
+          get_tx_root_validate_block_hash(
+            tx,
+            bad_state_queue_node_ref_input_index,
+            ct_token_asset_name,
+          )
+
+        // -------- Min fee: validation-specific logic START --------
+        // Membership proof for tx
+        expect
+          validate_membership_proof(
+            tx,
+            plutarch_midgard_merkle_validator_inclusion,
+            MembershipProofInputs {
+              tx_root: extracted_tx_root,
+              tx_hash: bad_tx_hash,
+              tx_info_hash: bad_tx_info_hash,
+              tx_to_root_proof_path: bad_tx_to_root_proof_path,
+            },
+          )
+
+        verify_hash_32(bad_tx, bad_tx_hash)
+        verify_hash_32(bad_tx_body, bad_tx.body)
+
+        // Fee must be greater than or equal to the minimum fee
+        expect bad_tx_body.fee < get_min_transaction_fee(bad_tx)
+
+        // -------- Zero input: validation-specific logic END --------
+        expect
+          validate_final_step_output(
+            tx,
+            index_output,
+            fraud_proof_token_policy,
+            ct_token_asset_name,
+            step_datum,
+            computation_thread_redeemer_index,
+          )
+
+        True
+      }
+    }
+  }
+
+  else(_) {
+    fail
+  }
+}
+
+//TODO: This will need execution traces to calculate it
+fn get_min_transaction_fee(_transaction: MidgardTxCompact) {
+  0
```

Comment:
We should use `todo` so that the compiler can catch it whenever we re-include failure on warnings in our CI.
```suggestion
  todo
```

Reply:
Oh we have `-D` included in our CI. @GeorgeFlerovsky what do you think if we remove it for the time being?